src/main/scala/analysis/Local.scala

```diff
@@ -0,0 +1,351 @@
+            (c, p) =>
+              graph.mergeCells(c, p._1)
+          }
+      }
+      val node = cell.node.get
+      node.flags.unknown = true
+      val test = graph.collapseNode(node)
+  }
+
+  def visit(n: CFGPosition): Unit = {
+    if visited.contains(n) then
+      return
+    else
+      visited.add(n)
+    n match
+      case DirectCall(proc, target, label) if proc.name == "malloc" => // R0 = Malloc()
+        val size: BigInt = evaluateExpression(mallocRegister, constProp(n)) match
+          case Some(value) => value.value
+          case None => 0
+        val node = DSN(Some(graph), size)
+        node.allocationRegions.add(HeapLocation(nextMallocCount, proc, size))
+        node.flags.heap = true
+        graph.mergeCells(graph.varToCell(n)(mallocRegister)._1, node.cells(0))
+      case call: DirectCall if params.contains(call.target) => // Rx, Ry, ... Rn = FunctionCall()
+        // create call sites for the callees
+        val cs = CallSite(call, graph)
+        graph.callsites.add(cs)
+        cs.paramCells.foreach{
+          case (variable: Variable, slice: Slice) =>
+            visitPointerArithmeticOperation(call, graph.adjust(slice), variable, 0)
+        }
+        cs.returnCells.foreach{
+          case (variable: Variable, slice: Slice) =>
+            val returnArgument  = graph.varToCell(n)(variable)
+            graph.mergeCells(graph.adjust(returnArgument), graph.adjust(slice))
+        }
+      case Assign(variable: Variable, rhs: Expr, maybeString) =>
+        val expr: Expr = unwrapPaddingAndSlicing(rhs)
+        val lhsCell = graph.adjust(graph.varToCell(n)(variable))
+        var global = isGlobal(rhs, n)
+        var stack = isStack(rhs, n)
+        if global.isDefined then // Rx = global address
+          graph.mergeCells(lhsCell, global.get)
+        else if stack.isDefined then // Rx = stack address
+          graph.mergeCells(lhsCell, stack.get)
+        else
+          expr match
+            case BinaryExpr(op, arg1: Variable, arg2) => // Rx = Rx + c
+              val arg2Offset = evaluateExpression(arg2, constProp(n))
+              if op.equals(BVADD) && arg1.equals(stackPointer)
+              && arg2Offset.isDefined && isNegative(arg2Offset.get) then
+                () // the stack is handled prior to this
+//                val size = bv2SignedInt(arg2Offset.get)
+//                val node = DSN(Some(graph))
+//                node.allocationRegions.add(StackLocation("Stack_"+proc.name, proc, -size))
+//                node.flags.stack = true
+//                graph.mergeCells(lhsCell, node.cells(0))
+              else if /*varToSym.contains(n) &&  varToSym(n).contains(arg1) && */ arg2Offset.isDefined then
+                // merge lhs with cell(s) corresponding to (arg1 + arg2) where arg1 is cell and arg2 is an offset
+                val offset = evaluateExpression(arg2, constProp(n)).get.value
+                visitPointerArithmeticOperation(n, lhsCell, arg1, 0, false, offset)
```

Comment:
This is incorrectly assuming that the operation is BVADD.

Reply:
I think this is still an issue - visitPointerArithmeticOperation assumes that the operation is addition and this is not a case that has collapse = true

Reply:
Sorry I made changes locally to check for op.bvadd and marked this as resolved. It is enough to analyse some complex examples. The other operations are unclear specially, since one of the operands is unknown. I'm not sure if can represent subtractions soundly since we can't have negative offsets and we have to be able to handle the program flow insensitively